### PR TITLE
File Scoped Namespaces

### PR DESCRIPTION
**Pull request by Kahbazi**
I have only change the main project for now. Should I change the other projects too in this PR or wait for this one to be merged first?

Contributes to https://github.com/microsoft/reverse-proxy/issues/1351

### CODE REVIEW

**Comment by Tratcher** [comment]
@Kahbazi we're going to look at #1351 in triage before we decide to go forward with this. We'll get back to you next week.

**Comment by WeihanLi** [comment]
Would like to suggest adding namespace style config into the `.editorconfig` for better experience <<https://github.com/WeihanLi/SparkTodo/blob/01aeada4fa7107fe864696b8e06db480b73465a7/.editorconfig#L32>>

```
csharp_style_namespace_declarations=file_scoped:suggestion
```

**Comment by karelz** [comment]
Triage: We are ok with it. We want to keep samples as they are for now to not force sample users to install .NET 6 SDK when they copy-paste specific sample.

File: src/ReverseProxy/Yarp.ReverseProxy.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <RootNamespace>Yarp.ReverseProxy</RootNamespace>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <Nullable>enable</Nullable>
+    <LangVersion>10.0</LangVersion>
```
Comment (Tratcher) [inline src/ReverseProxy/Yarp.ReverseProxy.csproj:10]: This should be global:
https://github.com/microsoft/reverse-proxy/blob/b2932df6c37044f5a72a5ef9228cf43e6b64b858/Directory.Build.props#L9

File: test/ReverseProxy.Tests/Configuration/ConfigProvider/ConfigurationReadingExtensionsTests.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 using System.Collections.Generic;
 using Xunit;
+using Yarp.ReverseProxy.Configuration.ConfigProvider;
```
Comment (Tratcher) [inline test/ReverseProxy.Tests/Configuration/ConfigProvider/ConfigurationReadingExtensionsTests.cs:6]: ? There shouldn't be any changes that require this.

**Review by Tratcher — COMMENTED** [review]
Please update the other projects and tests as well. Only avoid projects in the samples and docs folders that target 3.1/5.0.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks @Kahbazi. I'll leave #1351 open for now and add notes about converting other parts of the repo.

**Pull request merged**